### PR TITLE
fix: resolves #6214

### LINKED ISSUE #6214: If the field will not be empty then why there is on-off button
**Describe the bug**

The on-off switch should not be there if the field is mandatory

**Screenshots**

Screenshot (118)

### PR DESCRIPTION
**Pull request by himanshujaidka**
Fixes #6214 

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by iamareebjamal** [comment]
This is not a fix. This is a workaround

**Comment by himanshujaidka** [comment]
> This is not a fix. This is a workaround

okay

**Comment by himanshujaidka** [comment]
Any hints?

**Comment by iamareebjamal** [comment]
Why are sponsors being created when the switch is off

**Comment by himanshujaidka** [comment]
> Why are sponsors being created when the switch is off

Yeah I know that but I want to ask that ay hints regarding how to resolve that

**Comment by iamareebjamal** [comment]
That is the hint. Find out why first, and prevent that to solve the issue

**Comment by himanshujaidka** [comment]
> That is the hint. Find out why first, and prevent that to solve the issue

okay,

**Comment by himanshujaidka** [comment]
@iamareebjamal sir, have a look in it

**Comment by iamareebjamal** [comment]
Now, if I enable the sponsor button?

**Comment by himanshujaidka** [comment]
> Now, if I enable the sponsor button?

then you have to add sponsor

**Comment by himanshujaidka** [comment]
> > Now, if I enable the sponsor button?
> 
> then you have to add sponsor

if you want to remove then the remove button is there, after removing go ahead with registration

**Comment by iamareebjamal** [comment]
If I enable sponsors, a sponsor should be added automatically. This was the previous behaviour. We have to fix the bug without changing previous behaviour

**Comment by himanshujaidka** [comment]
> If I enable sponsors, a sponsor should be added automatically. This was the previous behaviour. We have to fix the bug without changing previous behaviour

Do you mean that I don't want to delete the previous code?

**Comment by himanshujaidka** [comment]
https://user-images.githubusercontent.com/58654018/104132374-d8adac80-53a2-11eb-829a-23fc088b2b30.mp4

But I think it is working fine

**Comment by iamareebjamal** [comment]
> If I enable sponsors, a sponsor should be added automatically.

**Comment by iamareebjamal** [comment]
> Do you mean that I don't want to delete the previous code?

Never said that

**Comment by himanshujaidka** [comment]
> > If I enable sponsors, a sponsor should be added automatically.

I think there might be confusion if the user set  the toggle to on then there will be an "add sponsor" button after clicking on that user will able to add sponsor details and if the user use the remove button then the user will able to delete the sponsor

**Comment by himanshujaidka** [comment]
> > If I enable sponsors, a sponsor should be added automatically.

So u mean that there should not be a button of " Add Sponsor"?

**Comment by iamareebjamal** [comment]
Again, never said that. I simply said, the behaviour should be like it was before this PR, which was "If I enable sponsors, a sponsor should be added automatically."

@maze-runnar @sachinchauhan2889 Can you please explain clearly? I believe I'm not that good at communication

**Comment by maze-runnar** [comment]
@himanshujaidka when you turn on button the sponsor section already should be there as previously like this - 
Screenshot from 2021-01-11 11-59-39

But in you case i turn on button, then a button appears, then I have to click another button. That shouldn't happen at all
Screenshot from 2021-01-11 12-00-58

As @iamareebjamal said you have to fix the bug without disturbing any other behaviour. And I have no more hint than this.

**Comment by himanshujaidka** [comment]
> @himanshujaidka when you turn on button the sponsor section already should be there as previously like this -
> Screenshot from 2021-01-11 11-59-39
> 
> But in you case i turn on button, then a button appears, then I have to click another button. That shouldn't happen at all
> Screenshot from 2021-01-11 12-00-58
> 
> As @iamareebjamal said you have to fix the bug without disturbing any other behaviour. And I have no more hint than this.

I got that the, I am not supposed to change the behavior of the app thanks 👍🏻

File: app/components/forms/wizard/sponsors-step.js [hunk 1]
```diff
@@ -54,9 +54,8 @@ export default Component.extend(FormMixin, EventWizardMixin, {
       sponsor.deleteRecord();
     }
   },
-
```
Comment (iamareebjamal) [inline app/components/forms/wizard/sponsors-step.js:57]: Add line back

Comment (himanshujaidka) [reply app/components/forms/wizard/sponsors-step.js:57]: okay

**Comment by himanshujaidka** [comment]
> Not working

okay

**Comment by iamareebjamal** [comment]
@himanshu010 Please complete or close

**Comment by himanshujaidka** [comment]
> @himanshu010 Please complete or close

Okay today i will

**Pull request closed without merging**